src/coreclr/tools/Common/TypeSystem/Common/MetadataFieldLayoutAlgorithm.cs

```diff
@@ -134,6 +134,7 @@ out instanceByteSizeAndAlignment
                 return result;
             }
 
+            var layoutMetadata = type.GetClassLayout();
```

Comment:
This is very subtle. Maybe now that we need to look at field offsets from metadata even for non explicit layout types, `layoutMetadata` containing the `FieldAndOffset[]` outlived it's usefulness - maybe we should drop that field and add a first class `virtual LayoutInt MetadataOffset` property in `FieldDesc.Layout.cs`. Then we can check it in each layouting strategy as we're doing layout.

Reply:
(I can sketch that out in a .diff or push it to the branch tomorrow if it helps.)

Reply:
I'll take a shot at it before the end of the day and we can see what we think of it.

Reply:
> I'll take a shot at it before the end of the day and we can see what we think of it.

This turned out pretty nice, thank you!